panther_battery/src/dual_battery_publisher.cpp

```diff
@@ -0,0 +1,173 @@
+#include <panther_battery/dual_battery_publisher.hpp>
+
+#include <memory>
+#include <stdexcept>
+#include <utility>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <sensor_msgs/msg/battery_state.hpp>
+
+#include <panther_battery/battery.hpp>
+#include <panther_battery/battery_publisher.hpp>
+
+namespace panther_battery
+{
+
+DualBatteryPublisher::DualBatteryPublisher(
+  std::shared_ptr<rclcpp::Node> node, std::shared_ptr<Battery> & battery_1,
+  std::shared_ptr<Battery> & battery_2)
+: BatteryPublisher(node), battery_1_(std::move(battery_1)), battery_2_(std::move(battery_2))
+{
+  battery_pub_ = node_->create_publisher<BatteryStateMsg>("battery", 5);
+  battery_1_pub_ = node_->create_publisher<BatteryStateMsg>("battery_1_raw", 5);
+  battery_2_pub_ = node_->create_publisher<BatteryStateMsg>("battery_2_raw", 5);
+}
+
+void DualBatteryPublisher::Update()
+{
+  const auto header_stamp = node_->get_clock()->now();
+  battery_1_->Update(header_stamp, ChargerConnected());
+  battery_2_->Update(header_stamp, ChargerConnected());
+}
+
+void DualBatteryPublisher::Reset()
+{
+  const auto header_stamp = node_->get_clock()->now();
+  battery_1_->Reset(header_stamp);
+  battery_2_->Reset(header_stamp);
+}
+
+void DualBatteryPublisher::PublishBatteryState()
+{
+  try {
+    const auto battery_msg =
+      MergeBatteryMsgs(battery_1_->GetBatteryMsg(), battery_2_->GetBatteryMsg());
+    battery_pub_->publish(battery_msg);
+    BatteryStatusLogger(battery_msg);
+  } catch (const std::runtime_error & err) {
+    RCLCPP_WARN(
+      node_->get_logger(), "Failed to merge battery_1 and battery_2 messages: %s", err.what());
+    RCLCPP_WARN(node_->get_logger(), "Battery message will not be published");
+  }
```

Comment:
```suggestion
  const auto battery_msg =
    MergeBatteryMsgs(battery_1_->GetBatteryMsg(), battery_2_->GetBatteryMsg());
  battery_pub_->publish(battery_msg);
  BatteryStatusLogger(battery_msg);
```
Since all of the exceptions are impossible to take place, no exception will ever take place